packages/sdk/src/utils/Utils.ts

```diff
@@ -35,20 +35,42 @@ export class Utils {
 
     let cmciResource = `/${CicsCmciConstants.CICS_SYSTEM_MANAGEMENT}/${resourceName}/${cicsPlex}${region}`;
 
-    if (options != null) {
-      if (options.criteria != null && options.criteria.length > 0) {
-        const addParentheses = options.criteria.charAt(0) !== '(';
+    if (options && options.criteria) {
+      cmciResource += `${delimiter}${CicsCmciConstants.CRITERIA}=${this.enforceParentheses(encodeURIComponent(options.criteria))}`;
+      delimiter = "&";
+    }
 
-        cmciResource = cmciResource + delimiter + "CRITERIA=" + (addParentheses ? "(": "") +
-        encodeURIComponent(options.criteria) + (addParentheses ? ")": "") ;
-        delimiter = "&";
-      }
+    if (options && options.parameter) {
+      cmciResource += `${delimiter}PARAMETER=${encodeURIComponent(options.parameter)}`;
+      delimiter = "&";
+    }
 
-      if (options.parameter != null && options.parameter.length > 0) {
-        cmciResource = cmciResource + delimiter + "PARAMETER=" + encodeURIComponent(options.parameter);
-      }
+    if (options && options.queryParams && options.queryParams.summonly) {
+      cmciResource += `${delimiter}${CicsCmciConstants.SUMM_ONLY}`;
+      delimiter = "&";
+    }
+
+    if (options && options.queryParams && options.queryParams.nodiscard) {
```

Comment:
there could be some savings here too 😋 
```suggestion
    if (options?.queryParams?.nodiscard) {
```